rclpy/rclpy/executors.py

```diff
@@ -151,9 +151,8 @@ def __init__(self, *, context: Context = None) -> None:
         self._tasks: List[Tuple[Task, Optional[WaitableEntityType], Optional[Node]]] = []
         self._tasks_lock = Lock()
         # This is triggered when wait_for_ready_callbacks should rebuild the wait list
-        gc, gc_handle = _rclpy.rclpy_create_guard_condition(self._context.handle)
-        self._guard_condition = gc
-        self._guard_condition_handle = gc_handle
+        self._guard = GuardCondition(
+            callback=None, callback_group=None, context=self._context)
```

Comment:
Reusing `GuardCondition` to reuse executor code that uses  `guard_condition.handle`